Source/EventParser.swift

```diff
@@ -64,15 +70,20 @@ class EventParser {
     }
 
     private func dispatchEvent() {
-        guard !data.isEmpty
-        else { return }
-        let messageEvent = MessageEvent(data: data, lastEventId: lastEventId)
         if let lastEventId = lastEventId {
             connectionHandler.setLastEventId(lastEventId)
         }
-        handler.onMessage(eventType: eventType, messageEvent: messageEvent)
+        guard !data.isEmpty
+        else {
```

Comment:
What is the advantage of writing it this way, instead of the shorter:

```swift
if data.isEmpty {
    eventType == ""
    return
}
```
?

Reply:
There's a tendency to use `guard` over `if` in Swift if when the condition fails the code block should be exited. The compiler will enforce that the block is exited from the body of the `guard`. Practically, it doesn't matter in a simple condition like this, but the `guard` condition can introduce a binding that is valid in the body following the `guard` and as such must verify that the following body is only entered if the binding succeeds.